llama-index-integrations/llms/llama-index-llms-vertex/llama_index/llms/vertex/utils.py

```diff
@@ -105,6 +115,14 @@ async def _completion_with_retry(**kwargs: Any) -> Any:
 
             generation = client.start_chat(history=history)
             generation_config = dict(kwargs)
+            if datastore:
+                from vertexai.preview.generative_models import Tool, grounding
+                tool = Tool.from_retrieval(
+                    grounding.Retrieval(grounding.VertexAISearch(datastore=datastore))
```

Comment:
Interesting. Do you mind giving me a quick rundown of what this does to API calls? Does it force the LLM to act as an agent with this single retrieval tool?

Does vertexai support tools for any LLM? This would be really nice for agents for example, similar to openais tools api

Reply:
Hello Logan,

Thanks for your answer. What this does is, in case you have a Verte datastore defined (see this [link](https://cloud.google.com/dialogflow/vertex/docs/concept/data-store)) you can ground your model in the data store. This is incredibly convenient for RAG applications, as creating a datastore is very easy.

And it doesn't force the LLM to act as an agent. You are just giving the LLM your datastore as a resource. You can tell in the prompt to give an answer strictly using the datastore, to reduce hallucinations.